hyperspy/axes.py

```diff
@@ -1540,8 +1540,11 @@ def __getitem__(self, y):
         return ans
 
     def _axes_getter(self, y):
-        if y in self._axes:
-            return y
+        if isinstance(y, BaseDataAxis):
+            if y in self._axes:
+                return y
+            else:
+                raise ValueError(f"{y} is not in {self}")
```

Comment:
CAn you get this line covered by a test?

Reply:
Done.